ui/tests/acceptance/recovery/snapshots-test.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { module, test } from 'qunit';
+import { visit, currentRouteName, currentURL } from '@ember/test-helpers';
+import { setupApplicationTest } from 'ember-qunit';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { overrideResponse } from 'vault/tests/helpers/stubs';
+import { login } from 'vault/tests/helpers/auth/auth-helpers';
+import { GENERAL } from 'vault/tests/helpers/general-selectors';
+import { Response } from 'miragejs';
+
+module('Acceptance | recovery snapshots', function (hooks) {
+  setupApplicationTest(hooks);
+  setupMirage(hooks);
+
+  hooks.beforeEach(function () {
+    this.server.get('/sys/storage/raft/configuration', () => this.server.create('configuration', 'withRaft'));
+
+    this.store = this.owner.lookup('service:store');
+    return login();
+  });
+
+  test('it renders empty state when no snapshots are loaded', async function (assert) {
+    this.server.get('/sys/storage/raft/snapshot-load', () => {
+      return new Response(404, { 'Content-Type': 'application/json' }, JSON.stringify({ errors: [] }));
+    });
+
+    await visit('/vault/recovery/snapshots');
+
+    assert.strictEqual(currentURL(), '/vault/recovery/snapshots');
+
+    assert.dom(GENERAL.emptyStateTitle).hasText('Upload a snapshot to get started');
+    assert.dom(GENERAL.emptyStateActions).hasText('Upload snapshot');
+  });
+
+  test('it redirects to snapshot route when a snapshot is loaded', async function (assert) {
+    this.server.get('/sys/storage/raft/snapshot-load', () =>
+      overrideResponse(null, {
+        data: {
+          keys: ['1234'],
+        },
+      })
+    );
```

Comment:
You don't need `overrideResponse` here, that's just a convenience function to stub the response shape - which for some reason didn't work above!
```suggestion
    this.server.get('/sys/storage/raft/snapshot-load', () => {
    return {     
       data: {
          keys: ['1234'],
        },
    }
  })
```